From f0d7ef94124e1d4ee91ea27b8dbcb1a1753d4c97 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Sun, 13 Aug 2017 00:52:03 -0700
Subject: [PATCH 2588/2848] HADOOP-14678. AdlFilesystem#initialize swallows
 exception when getting user name. Contributed by
 John Zhuge.

(cherry picked from commit f735ad1b67ed82d9b11b1afd7ae39035a6aed18b)

Change-Id: I2fc0dd8090ac705c22a5803f2a05fb28cde8ad49
---
 .../org/apache/hadoop/fs/adl/AdlFileSystem.java    |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-tools/hadoop-azure-datalake/src/main/java/org/apache/hadoop/fs/adl/AdlFileSystem.java b/hadoop-tools/hadoop-azure-datalake/src/main/java/org/apache/hadoop/fs/adl/AdlFileSystem.java
index 56bd72a..f04eabe 100644
--- a/hadoop-tools/hadoop-azure-datalake/src/main/java/org/apache/hadoop/fs/adl/AdlFileSystem.java
+++ b/hadoop-tools/hadoop-azure-datalake/src/main/java/org/apache/hadoop/fs/adl/AdlFileSystem.java
@@ -128,6 +128,8 @@ public void initialize(URI storeUri, Configuration conf) throws IOException {
       userName = UserGroupInformation.getCurrentUser().getShortUserName();
     } catch (IOException e) {
       userName = "hadoop";
+      LOG.warn("Got exception when getting Hadoop user name."
+          + " Set the user name to '" + userName + "'.", e);
     }
 
     this.setWorkingDirectory(getHomeDirectory());
-- 
1.7.9.5

